tests/ui/dyn-star/llvm-old-style-ptrs.rs

```diff
@@ -1,8 +1,6 @@
 // run-pass
 // compile-flags: -Copt-level=0 -Cllvm-args=-opaque-pointers=0
```

Comment:
This test can be deleted entirely. I guess that can be done as part of rustc_codegen_llvm cleanups to drop typed pointer handling (which will probably break this tests).

Reply:
There's a related line here:
https://github.com/rust-lang/rust/blob/9339f446a5302cd5041d3f3b5e59761f36699167/compiler/rustc_codegen_ssa/src/base.rs#L274

And yeah, I think dropping typed pointers entirely can be a later followup.